streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -491,25 +493,23 @@ public void testStateThreadClose() throws Exception {
                 () -> streams.localThreadsMetadata().stream().allMatch(t -> t.threadState().equals("DEAD")),
                 "Streams never stopped"
             );
-        } finally {
             streams.close();
```

Comment:
Thanks @mjsax for review. Will do.

Reply:
@mjsax We should not remove this line, otherwise the line 498(original) will throw a timeout exception.
The streams state will be always **RUNNING** if we skip `close()`

Reply:
As `KafkaStreams` implements the `AutoCloseable` interface now, `close()` should be called automatically when the `try {}` block is left -- that is the whole purpose of `AutoClosable` and try-with-resource construct -- it frees you up to call `close()` explicitly (so you cannot forget any longer).

Reply:
I missed the fact that we moved the `waitForCondition` check _inside_ of the try-catch block... For this case, we need to call `close` explicitly of course, as we are still in the block and `close()` is not auto-called yet...

Sorry for the confusion.